mgr/dashboard: Fix `npm run fixmod` command

p-na opened the pull request:
As the previous implementation didn't work for `spec.ts` files.

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

**Explanation**

`npm run fixmod` does the same as `npm run fix`, but quicker. This is due to it only passing files to `prettier` which actually have been changed (but not yet committed). I use it for quickly fixing frontend files before making a commit.

Revelant for developers only.

*edit*

For the record: `prettier` does not accept a list of files to be processed, just one file or a single pattern, which is hard to implement for "npm scripts", therefore I adapted and replaced this solution to use a new package which does the same thing.

<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

s0nea commented:
jenkins test make check

tchaikov commented:
retest this please.

s0nea commented:
jenkins test dashboard

tchaikov commented:
retest this please.

ricardoasmarques commented:
jenkins test make check arm64

votdev commented:
I tried this PR by running the command without changing a file, but the tool reported 42 changed files. How does this happen?
```
╰─# npm run fixmod

> ceph-dashboard@0.0.0 fixmod /ceph/src/pybind/mgr/dashboard/frontend
> pretty-quick

🔍  Finding changed files since git revision bc49c2ec85.
🎯  Found 42 changed files.
✅  Everything is awesome!
```
After modifying only one JS file the tool is reporting the following:
```
🔍  Finding changed files since git revision bc49c2ec85.
🎯  Found 43 changed files.
✅  Everything is awesome!
```
OK, it noticed the modified file, but where did the others came from?

votdev commented:
> I tried this PR by running the command without changing a file, but the tool reported 42 changed files. How does this happen?
> 
> ```
> ╰─# npm run fixmod
> 
> > ceph-dashboard@0.0.0 fixmod /ceph/src/pybind/mgr/dashboard/frontend
> > pretty-quick
> 
> 🔍  Finding changed files since git revision bc49c2ec85.
> 🎯  Found 42 changed files.
> ✅  Everything is awesome!
> ```
> 
> After modifying only one JS file the tool is reporting the following:
> 
> ```
> 🔍  Finding changed files since git revision bc49c2ec85.
> 🎯  Found 43 changed files.
> ✅  Everything is awesome!
> ```
> 
> OK, it noticed the modified file, but where did the others came from?

I think these are the compiled JS files that are generated when running ``npm start``, but aren't they ignored by ``.gitignore``?

p-na commented:
@votdev Thanks for the thorough review! I found out that the default branch `pretty-quick` compares with is `master`. It seems I had an updated master, so that it looked correct for me. Your repository was a few commits behind master. I fixed this. I also added the same pattern to be used to find the files as we use for `prettier`.

votdev commented on src/pybind/mgr/dashboard/frontend/package.json:
```diff
@@ -20,7 +20,7 @@
     "fix:prettier": "prettier --write \"{src,e2e}/**/*.{ts,scss}\"",
     "fix:tslint": "npm run lint:tslint -- --fix",
-    "fixmod": "prettier --write $(git rev-parse --show-toplevel)/$(git status --porcelain | grep -E '^(\\sM|\\?\\?|A)' | sed -e 's/^...//' | grep -E '^(src|e2e).*\\.(ts|scss)$') 1>/dev/null 2>&1 && echo 'done' || echo 'no changes found'",
+    "fixmod": "pretty-quick --pattern \"{src,e2e}/**/*.{ts,scss}\" --branch HEAD",
```
Would it be possible to add the ``--verbose`` know, i would like to see which files are processed.

tchaikov commented:
- http://pulpito.ceph.com/kchai-2019-06-09_04:37:10-rados-wip-kefu-testing-2019-06-09-0009-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-06-11_01:51:33-rados-wip-kefu-testing-2019-06-09-0009-distro-basic-smithi/

test failures tracked by 

- http://tracker.ceph.com/issues/17882
- http://tracker.ceph.com/issues/40120
- the misdirected op issue is not related to this issue.
